### PR TITLE
Command runner

### PR DESCRIPTION
**Pull request by zhengyangyong**
JAV-252&305 add SecKillRunner load admin service promotion creation then start , and integration test for whole check

### CODE REVIEW

File: seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  private static final Logger logger = LoggerFactory.getLogger(SecKillRunner.class);
+
+  private final PromotionRepository promotionRepository;
+  private final CouponEventRepository eventRepository;
+  private final List<SecKillCommandService<String>> commandServices;
+  private final List<SecKillPersistentRunner<String>> persistentRunners;
+  private final AtomicInteger claimedCoupons = new AtomicInteger();
+
+  public SecKillRunner(
```
Comment (seanyinx) [inline seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java]: the class name is not revealing its intention

this class shall be generic

File: seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    Iterable<Promotion> promotions = promotionRepository.findAll();
+    for (Promotion promotion : promotions) {
+      if (promotion.getPublishTime().before(new Date())) {
+        return promotion;
+      }
+    }
+    return null;
+  }
+
+  public void run() {
```
Comment (seanyinx) [inline seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java]: this method is too long

File: seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    CompletableFuture.runAsync(() -> {
+      boolean promotionLoaded = false;
+      while (!Thread.currentThread().isInterrupted() && !promotionLoaded) {
```
Comment (seanyinx) [inline seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java]: why rely on if promotion loaded?

File: integration-test/src/test/java/io/servicecomb/poc/demo/seckill/SecKillIntegrationTestApplicationTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = IntegrationTestApplication.class)
+@WebAppConfiguration
+@AutoConfigureMockMvc
+public class SecKillIntegrationTestApplicationTest {
```
Comment (seanyinx) [inline integration-test/src/test/java/io/servicecomb/poc/demo/seckill/SecKillIntegrationTestApplicationTest.java]: the class name looks a bit weird

File: seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+public class SecKillRunner {
+
+  private static final Logger logger = LoggerFactory.getLogger(SecKillRunner.class);
+
+  private final PromotionRepository promotionRepository;
+  private final CouponEventRepository eventRepository;
+  private final List<SecKillCommandService<String>> commandServices;
+  private final List<SecKillPersistentRunner<String>> persistentRunners;
+  private final AtomicInteger claimedCoupons = new AtomicInteger();
```
Comment (seanyinx) [inline seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java]: claimed coupons shall be local variable

Comment (zhengyangyong) [reply seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java]: Done

File: seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java (hunk 1 above)
Comment (zhengyangyong) [reply seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java]: Change to SecKillPromotionLancher or SecKillPromotionBootstrap ?

File: seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java (hunk 2 above)
Comment (zhengyangyong) [reply seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java]: Done

File: seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java (hunk 3 above)
Comment (zhengyangyong) [reply seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillRunner.java]: only load any one promotion temporary

File: integration-test/src/test/java/io/servicecomb/poc/demo/seckill/SecKillIntegrationTestApplicationTest.java (hunk 4 above)
Comment (zhengyangyong) [reply integration-test/src/test/java/io/servicecomb/poc/demo/seckill/SecKillIntegrationTestApplicationTest.java]: Done

Comment (zhengyangyong) [reply integration-test/src/test/java/io/servicecomb/poc/demo/seckill/SecKillIntegrationTestApplicationTest.java]: Done

File: seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillPromotionBootstrap.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  public void run() {
+    CompletableFuture.runAsync(() -> {
+      int startedPromotionId = 0;
+      while (!Thread.currentThread().isInterrupted()) {
+        try {
+          Iterable<Promotion> promotions = promotionRepository.findByIdGreaterThan(startedPromotionId);
+          for (Promotion promotion : promotions) {
+            startUpPromotion(promotion);
```
Comment (seanyinx) [inline seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillPromotionBootstrap.java:55]: it seems the test for this part is missing

Comment (zhengyangyong) [reply seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillPromotionBootstrap.java:55]: SecKillIntegrationTest covered，but I don't know why jacoco missing it

**Pull request merged**